File: SqlLite/app/src/main/java/com/example/sqllite/MainActivity.java
package com.example.sqllite;

import android.app.Activity;
import android.content.Context;
import android.content.DialogInterface;
import android.content.Intent;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.support.v7.app.AlertDialog;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.Button;
import android.widget.ListView;
import android.widget.TextView;
import android.widget.Toast;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    ListView listView;
    Button tambah_data;
    customAdapter adapter;
    Cursor cursor;
    Mahasiswa mahasiswa;
    ArrayList<Objek> list;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        listView=findViewById(R.id.listview);
        tambah_data=findViewById(R.id.tambah);

        mahasiswa=new Mahasiswa(getApplicationContext());

        tambah_data.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent=new Intent(getApplicationContext(),tambah_data.class);
                startActivity(intent);
            }
        });
        ambil_data();
    }
    void ambil_data(){
        list=new ArrayList<Objek>();
        cursor=mahasiswa.tampil_data();
        if(cursor!=null && cursor.getCount()>0){
            while (cursor.moveToNext()){
                String id=cursor.getString(0);
                String nama=cursor.getString(1);
                String alamat=cursor.getString(2);
                list.add(new Objek(
                        id,
                        nama,
                        alamat
                ));
            }
            adapter=new customAdapter(getApplicationContext(),list,MainActivity.this);
            listView.setAdapter(adapter);
        }
    }

    @Override
    protected void onResume() {
        super.onResume();
        ambil_data();
    }
}

class customAdapter extends BaseAdapter {
    Activity activity;
    Context context;
    LayoutInflater layoutInflater;
    ArrayList<Objek> model;
    Mahasiswa mahasiswa;

    customAdapter(Context context,ArrayList<Objek>model,Activity activity){
        this.context=context;
        this.model=model;
        this.activity=activity;
        layoutInflater = LayoutInflater.from(context);
        mahasiswa=new Mahasiswa(context);
    }

    @Override

    //hitung besar array list
    public int getCount() {
        return model.size();
    }

    @Override
    //cari posisi array list
    public Object getItem(int i) {
        return model.get(i);
    }

    @Override
    //cari index dlm array list
    public long getItemId(int i) {
        return i;
    }

    TextView id,nama,alamat;
    Button edit,hapus;
    @Override
    public View getView(int i, View view, ViewGroup viewGroup) {
        View view1=layoutInflater.inflate(R.layout.list_data,viewGroup,false);

        id=view1.findViewById(R.id.id_data);
        nama=view1.findViewById(R.id.nama);
        alamat=view1.findViewById(R.id.alamat);
        edit=view1.findViewById(R.id.edit);
        hapus=view1.findViewById(R.id.hapus);

        id.setText(model.get(i).getId());
        nama.setText(model.get(i).getNama());
        alamat.setText(model.get(i).getAlamat());

        edit.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent=new Intent(context,Edit.class);
                intent.putExtra("nama",model.get(i).getNama());
                intent.putExtra("alamat",model.get(i).getAlamat());
                intent.putExtra("id",model.get(i).getId());
                intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
                context.startActivity(intent);
            }
        });
        hapus.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                AlertDialog.Builder builder=new AlertDialog.Builder(activity);
                builder.setTitle("Tanya");
                builder.setMessage("Apakah ingin dihapus?");
                builder.setPositiveButton("Hapus", new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialogInterface, int i) {
                        mahasiswa.delete_data(model.get(i).getId());
//                        ((Activity)context).overridePendingTransition(0,0);

                        Intent intent=new Intent(context,MainActivity.class);
                        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
                        context.startActivity(intent);
//                        ((Activity)context).finish();
                    }
                });
                builder.setNegativeButton("Tidak", new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialogInterface, int i) {
                        dialogInterface.dismiss();
                    }
                });
                AlertDialog alertDialog=builder.create();
                alertDialog.show();
            }
        });


        return view1;
    }
}

File: Kontak/app/src/main/java/com/example/kontak/MainActivity.kt
package com.example.kontak

import android.content.Context
import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.design.widget.FloatingActionButton
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.*
import com.android.volley.Response
import com.android.volley.toolbox.StringRequest
import com.android.volley.toolbox.Volley
import org.json.JSONException
import org.json.JSONObject

class MainActivity : AppCompatActivity() {
    lateinit var tambahKontak:FloatingActionButton;
    lateinit var buttoncari:Button;
    lateinit var list:ArrayList<Objek>;
    lateinit var listview:ListView;
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        tambahKontak=findViewById(R.id.tambahkontak);
        buttoncari=findViewById(R.id.buttoncari);
        listview=findViewById(R.id.listview);
        tambahKontak.setOnClickListener(){
            var intent=Intent(this,TambahKontak::class.java);
            startActivity(intent);
        }
        buttoncari.setOnClickListener(){
            var intent=Intent(this,Cari::class.java);
            startActivity(intent);
        }
        load_data();

    }
    fun load_data(){
        var url="http://192.168.237.125/latihan/tampil_data.php";
        var stringreq=StringRequest(1,url,
        Response.Listener {response ->
//                Toast.makeText(this,response,Toast.LENGTH_SHORT).show();
            try {
                var jsonobject=JSONObject(response);
                var jsonArray=jsonobject.getJSONArray("hasil");
                var list=ArrayList<Objek>();
                for(i in 0 until jsonArray.length()){
                    var objek=jsonArray.getJSONObject(i);
                    var id:String=objek.getString("id");
                    var nama:String=objek.getString("nama");
                    var nim:String=objek.getString("nim");
                    list.add(Objek(id,nama,nim));
                }
                var adapter=custom_adapter(this,list);
                listview.adapter=adapter;
            }
            catch (e:JSONException){

            }
        },
        Response.ErrorListener {error ->
                 Toast.makeText(this,"Gagal",Toast.LENGTH_SHORT).show();
        });
        var reqQue=Volley.newRequestQueue(this);
        reqQue.add(stringreq);
    }
}

class custom_adapter(var context:Context,var list:ArrayList<Objek>):BaseAdapter(){
    var layoutinflater=LayoutInflater.from(context);
    override fun getCount(): Int {
        return list.size;
    }

    override fun getItem(p0: Int): Any {
        return list[p0];
    }

    override fun getItemId(p0: Int): Long {
        return p0.toLong();
    }

    override fun getView(p0: Int, p1: View?, p2: ViewGroup?): View {
        var view=p1;
        if(view==null){
            view=layoutinflater.inflate(R.layout.list_item,p2,false);
        }
        lateinit var nama:TextView;
        lateinit var nim:TextView;
        nama=view!!.findViewById(R.id.hasilnama);
        nim=view!!.findViewById(R.id.hasilnim);
        nama.setText(list[p0].nama);
        nim.setText(list[p0].nim);
        return view;
    }

}